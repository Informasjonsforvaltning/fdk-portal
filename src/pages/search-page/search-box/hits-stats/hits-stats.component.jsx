import React from 'react';
import PropTypes from 'prop-types';
import localization from '../../../../lib/localization';
import { getConfig } from '../../../../config';

export const HitsStats = props => {
  const {
    countDatasets,
    countTerms,
    countApis,
    countInformationModels,
    filteringOrTextSearchPerformed
  } = props;

  // Do not show anything if we don't have the stats yet
  if (
    countDatasets === null ||
    countTerms === null ||
    countApis === null ||
    countInformationModels === null
  ) {
    return null;
  }

  let content = '';

  if (getConfig().themeNap) {
    let template;
    if (!filteringOrTextSearchPerformed) {
      template = localization.hitstats.nosearchNap;
    } else if (countDatasets === 0) {
      template = localization.hitstats.nohits;
    } else {
      template = localization.hitstats.searchNap;
    }

    content = localization.formatString(template, countDatasets);
  } else {
    const nohits =
      countDatasets === 0 &&
      countApis === 0 &&
      countTerms === 0 &&
      countInformationModels === 0;

    let template;
    if (!filteringOrTextSearchPerformed) {
      template = localization.hitstats.nosearch;
    } else if (nohits) {
      template = localization.hitstats.nohits;
    } else {
      template = localization.hitstats.search;
    }

    content = localization.formatString(
      template,
      countDatasets,
      countApis,
      countTerms,
      countInformationModels
    );
  }

  return (
    <div className="sk-hits-stats" data-qa="hits-stats">
      <div className="sk-hits-stats__info" data-qa="info">
        {content}
      </div>
    </div>
  );
};

HitsStats.defaultProps = {
  countDatasets: 0,
  countTerms: 0,
  countApis: 0,
  countInformationModels: 0,
  filteringOrTextSearchPerformed: false
};

HitsStats.propTypes = {
  countDatasets: PropTypes.number,
  countTerms: PropTypes.number,
  countApis: PropTypes.number,
  countInformationModels: PropTypes.number,
  filteringOrTextSearchPerformed: PropTypes.bool
};
