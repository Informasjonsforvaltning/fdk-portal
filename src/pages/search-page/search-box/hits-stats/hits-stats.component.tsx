import React, { FC } from 'react';

import localization from '../../../../lib/localization';

interface Props {
  countDatasets?: number;
  countTerms?: number;
  countApis?: number;
  countInformationModels?: number;
}

export const HitsStats: FC<Props> = ({
  countDatasets = 0,
  countTerms = 0,
  countApis = 0,
  countInformationModels = 0
}) => (
  <span>
    {localization.formatString(
      countDatasets === 0 &&
        countApis === 0 &&
        countTerms === 0 &&
        countInformationModels === 0
        ? localization.hitstats.nohits
        : localization.hitstats.search,
      countDatasets,
      countApis,
      countTerms,
      countInformationModels
    )}
  </span>
);
