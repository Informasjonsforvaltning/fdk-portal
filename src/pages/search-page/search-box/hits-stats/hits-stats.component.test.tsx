import React from 'react';
import { shallow } from 'enzyme';
import { HitsStats } from './hits-stats.component';

test('should render HitsStats correctly with hits ', () => {
  const defaultProps = {
    countDatasets: 100,
    countApis: 100,
    countTerms: 100,
    countInformationModels: 100
  };
  const wrapper = shallow(<HitsStats {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render HitsStats correctly when no hits', () => {
  const defaultProps = {
    countDatasets: 0,
    countApis: 0,
    countTerms: 0,
    countInformationModels: 0
  };
  const wrapper = shallow(<HitsStats {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
