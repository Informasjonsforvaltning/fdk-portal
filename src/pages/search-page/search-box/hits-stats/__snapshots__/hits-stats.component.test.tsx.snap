// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HitsStats correctly when no hits 1`] = `
<span>
  Sorry, no hits
</span>
`;

exports[`should render HitsStats correctly with hits  1`] = `
<span>
  Search gave 100 hit(s) among datasets, 100 hit(s) among APIs, 100 hit(s) among concepts 100 hit(s) among information models
</span>
`;
